lib/theme.json

```diff
@@ -191,7 +191,7 @@
 			"text": true
 		},
 		"shadow": {
-			"defaultPresets": true,
```

Comment:
@madhusudhand This is a breaking change for themes that have shadow presets with the same name as the core presets as it changes the CSS output for the theme. It should have a theme.json version bump and migration to maintain backwards compatibility.

I've been discussing this in https://github.com/WordPress/gutenberg/pull/58409#issuecomment-1997603910 since I want to change the default for the font sizes presets as well.

Reply:
> Meaning that `defaultPresets` are off by default and when appearanceTools is set to true, they are enabled, which makes existing themes that are using default presets to work as before.

`appearanceTools` is also `false` by default, so it's still a problem for themes that haven't configured either.

Before this change, with a theme that has custom `settings.shadow.presets` defined with matching slugs to those in the defailt lib/theme.json and `settings.shadow.defaultPresets` and `settings.appearanceTools` both undefined, you'll end up with different CSS that gets output to the page.

In the first case the default styles will be output and in the second case, the theme styles will be output.

Here's an example of what I'm talking about.

```jsonc
// Relevant parts of theme.json
{
  "settings": {
    // `appearanceTools` is not set
    "shadow": {
      // `defaultPresets` is not set
      "presets": [
        {
          "name": "Natural",
          "slug": "natural",
          "shadow": "2px 2px 3px #000"
        }
      ]
    }
  }
}
```

```css
/* Generated CSS before */
--wp--preset--shadow--natural: 6px 6px 9px rgba(0, 0, 0, 0.2);
```

```css
/* Generated CSS after */
--wp--preset--shadow--natural: 2px 2px 3px #000;
```

It's because to the `settings.shadow.defaultPresets` being used as the value for `prevent_override` in `PRESETS_METADATA` in `WP_Theme_JSON`. The setting defines the behavior of matching slugs in the default and theme origins for presets in theme.json.

https://github.com/WordPress/gutenberg/blob/cdfd75183952ce7ae1ec78e673ac719e002d2de8/lib/class-wp-theme-json-gutenberg.php#L181

https://github.com/WordPress/gutenberg/blob/cdfd75183952ce7ae1ec78e673ac719e002d2de8/lib/class-wp-theme-json-gutenberg.php#L2868-L2876

I'm suggesting that we revert this for 6.5 and use the mechanism to opt-in to breaking changes like this that I'm adding in #58409.

Reply:
Thank for details @ajlende. 
I see that it may break a shadow when the following conditions are met 
* `appearanceTools: false`
* theme defined custom shadows 
* and they are defined with same slug as defaultPresets

Also, prior to 6.5 shadows are enabled for `button` block and in global styles only.
It seems to be very unlikely edge case and shouldn't have big impact on themes.

@annezazu @swissspidy @t-hamano what do you think?

Reply:
Also cc @fabiankaegy editor triage co-lead.

Reply:
I decided it was best to create a dedicated issue where this can be discussed.

https://github.com/WordPress/gutenberg/issues/59989

This is also useful as it can be placed in `Triage` on the [WP 6.5 Editor board](https://github.com/orgs/WordPress/projects/148) for review by Editor triage leads.